File: <100/060/060.jl
using Primes

pairCat(n:: Int, m:: Int):: Int = parse(Int, string(n) * string(m))
arePairPrimes(n:: Int, m:: Int):: Bool = pairCat(n, m) |> isprime && pairCat(m, n) |> isprime 


function allPairPrimes(arr:: Array{Int}):: Bool
  L:: Int = arr |> length
  for j ∈ 1:(L-1)
    if arePairPrimes(arr[j], arr[L]) |> !
      return false
    end
  end

  return true
end


function naiveSolve(upper_limit:: Int):: Int
  Ps:: Array{Int} = primes(upper_limit)
  L:: Int = Ps |> length
  arr:: Array{Int} = []
  answer:: Int = 10^18

  for j1 ∈ 1:L
    push!(arr, Ps[j1])
    if arr |> sum >= answer
      pop!(arr)
      break
    end
    for j2 ∈ (j1+1):L
      push!(arr, Ps[j2])
      if arr |> sum >= answer
        pop!(arr)
        break
      end
      if arr |> allPairPrimes |> !
        pop!(arr)
        continue
      end
      for j3 ∈ (j2+1):L
        push!(arr, Ps[j3])
        if arr |> sum >= answer
          pop!(arr)
          break
        end
        if arr |> allPairPrimes |> !
          pop!(arr)
          continue
        end
        for j4 ∈ (j3+1):L
          push!(arr, Ps[j4])
          if arr |> sum >= answer
            pop!(arr)
            break
          end
          if arr |> allPairPrimes |> !
            pop!(arr)
            continue
          end
          for j5 ∈ (j4+1):L
            push!(arr, Ps[j5])
            if arr |> sum >= answer
              pop!(arr)
              break
            end
            if arr |> allPairPrimes |> !
              pop!(arr)
              continue
            end
            s:: Int = arr |> sum
            if s < answer
              answer = s
            end
            pop!(arr)
          end
          pop!(arr)
        end
        pop!(arr)
      end
      pop!(arr)
    end
    pop!(arr)
  end

  return answer
end


function main():: Nothing
  upper_limit:: Int = 10^4
  upper_limit |> naiveSolve |> println
end


@time main()

File: <100/060/060.py
import sympy
import time
import sys
sys.setrecursionlimit(10**8)


def concat(n: int, m: int) -> int:
    sn = str(n)
    sm = str(m)
    return int(sn + sm)


def allPairsPrimes(l: list[int], p: int) -> bool:
    for q in l:
        pq = concat(p, q)
        if not sympy.isprime(pq):
            return False
        qp = concat(q, p)
        if not sympy.isprime(qp):
            return False

    return True


def dfs(l: list[int], temporal) -> int:
    if len(l) == 6:
        s = sum(l)
        return s
    
    p = l[-1]
    while True:
        p = sympy.nextprime(p)
        if p > 9999:
            return temporal
        if not allPairsPrimes(l[1:], p):
            continue
        
        l.append(p)
        temporal = min(temporal, dfs(l, temporal))
        l.pop()


def main() -> int:
    l = [0]
    temporal = 10**8
    temporal = dfs(l, temporal)

    return temporal


if __name__ == '__main__':
    start = time.time()
    answer = main()
    elapsed = time.time() - start

    print(f'{elapsed} sec')
    print(answer)
